infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
It seems I'm always too optimistic about how long it will take to prepare a patch here.

`RelayContext` must have the `m` parameter, that's a given (see previous comment; `(Monad)IO` is not enough). But doing this means we can't give the ether `HasLens`, because that type variable is not in scope. I thought, no problem, I'll just define `MonadRelayMem` without mentioning ether, i.e. `askRelayMem :: m (RelayContext m)` is its sole member. But that's no good, because it prevents us from defining `MonadRelayMem WalletWebMode`: the wallet mode has a `RealModeContext`, and `RealModeContext` has a `RelayContext RealMode`, but we would need `RelayContext WalletWebMode`.

@georgeee @gromakovsky I know that problems like these are exactly why you're skeptical of having the `m` parameter. But concluding that we shouldn't have that parameter is essentially jamming a square peg into a round hole. The problem is the infrastructure which delivers the abstract relayer, not the abstract relayer itself.